collectors/ebpf.plugin/ebpf_vfs.c

```diff
@@ -0,0 +1,949 @@
+    vfs_publish_aggregated[NETDATA_KEY_PUBLISH_VFS_READ].ncall = res[NETDATA_KEY_CALLS_VFS_READ] +
+                                                             res[NETDATA_KEY_CALLS_VFS_READV];
+    vfs_publish_aggregated[NETDATA_KEY_PUBLISH_VFS_WRITE].ncall = res[NETDATA_KEY_CALLS_VFS_WRITE] +
+                                                              res[NETDATA_KEY_CALLS_VFS_WRITEV];
+    vfs_publish_aggregated[NETDATA_KEY_PUBLISH_VFS_FSYNC].ncall = res[NETDATA_KEY_CALLS_VFS_FSYNC];
+    vfs_publish_aggregated[NETDATA_KEY_PUBLISH_VFS_OPEN].ncall = res[NETDATA_KEY_CALLS_VFS_OPEN];
+    vfs_publish_aggregated[NETDATA_KEY_PUBLISH_VFS_CREATE].ncall = res[NETDATA_KEY_CALLS_VFS_CREATE];
+
+    vfs_publish_aggregated[NETDATA_KEY_PUBLISH_VFS_UNLINK].nerr = res[NETDATA_KEY_ERROR_VFS_UNLINK];
+    vfs_publish_aggregated[NETDATA_KEY_PUBLISH_VFS_READ].nerr = res[NETDATA_KEY_ERROR_VFS_READ] +
+                                                                res[NETDATA_KEY_ERROR_VFS_READV];
+    vfs_publish_aggregated[NETDATA_KEY_PUBLISH_VFS_WRITE].nerr = res[NETDATA_KEY_ERROR_VFS_WRITE] +
+                                                                 res[NETDATA_KEY_ERROR_VFS_WRITEV];
+    vfs_publish_aggregated[NETDATA_KEY_PUBLISH_VFS_FSYNC].nerr = res[NETDATA_KEY_ERROR_VFS_FSYNC];
+    vfs_publish_aggregated[NETDATA_KEY_PUBLISH_VFS_OPEN].nerr = res[NETDATA_KEY_ERROR_VFS_OPEN];
+    vfs_publish_aggregated[NETDATA_KEY_PUBLISH_VFS_CREATE].nerr = res[NETDATA_KEY_ERROR_VFS_CREATE];
+
+    vfs_aggregated_data[NETDATA_KEY_PUBLISH_VFS_WRITE].bytes = (uint64_t)res[NETDATA_KEY_BYTES_VFS_WRITE] +
+                                                               (uint64_t)res[NETDATA_KEY_BYTES_VFS_WRITEV];
+    vfs_aggregated_data[NETDATA_KEY_PUBLISH_VFS_READ].bytes = (uint64_t)res[NETDATA_KEY_BYTES_VFS_READ] +
+                                                              (uint64_t)res[NETDATA_KEY_BYTES_VFS_READV];
+}
+
+/**
+ * Sum PIDs
+ *
+ * Sum values for all targets.
+ *
+ * @param swap output structure
+ * @param root link list with structure to be used
+ */
+static void ebpf_vfs_sum_pids(netdata_publish_vfs_t *vfs, struct pid_on_target *root)
+{
+    uint32_t local_write_call = 0;
+    uint32_t local_writev_call = 0;
+    uint32_t local_read_call = 0;
+    uint32_t local_readv_call = 0;
+    uint32_t local_unlink_call = 0;
+    uint32_t local_fsync_call = 0;
+    uint32_t local_open_call = 0;
+    uint32_t local_create_call = 0;
+
+    uint64_t local_write_bytes = 0;
+    uint64_t local_writev_bytes = 0;
+    uint64_t local_read_bytes = 0;
+    uint64_t local_readv_bytes = 0;
+
+    uint32_t local_write_err = 0;
+    uint32_t local_writev_err = 0;
+    uint32_t local_read_err = 0;
+    uint32_t local_readv_err = 0;
+    uint32_t local_unlink_err = 0;
+    uint32_t local_fsync_err = 0;
+    uint32_t local_open_err = 0;
+    uint32_t local_create_err = 0;
```

Comment:
struct netdata_publish_vfs ?

Reply:
You are right, I did not use initially, because 16 bytes of the structure are never used, but I rethink about this, and I converted for structure. Thank you!